packages/@aws-cdk/aws-lambda/lib/lambda.ts

```diff
@@ -226,4 +257,38 @@ export class Function extends FunctionRef {
             variables: this.environment
         };
     }
+
+    /**
+     * If configured, set up the VPC-related properties
+     *
+     * Returns the VpcConfig that should be added to the
+     * Lambda creation properties.
+     */
+    private addToVpc(props: FunctionProps): cloudformation.FunctionResource.VpcConfigProperty | undefined {
+        if (!props.vpc) { return undefined; }
+
+        let securityGroup = props.securityGroup;
+        if (!securityGroup) {
+            securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
+                vpc: props.vpc,
+                description: 'Automatic security group for Lambda Function ' + this.uniqueId,
+            });
+        }
+
+        this._connections = new ec2.Connections({ securityGroup });
+
+        // Pick subnets, make sure they're not Public. Routing through an IGW
+        // won't work because the ENIs don't get a Public IP.
+        const subnets = props.vpc.subnets(props.vpcPlacement);
+        for (const subnet of subnets) {
+            if (props.vpc.publicSubnets.indexOf(subnet) > -1) {
+                throw new Error('Not possible to place Lambda Functions in a Public subnet');
```

Comment:
I can't mention Internal because in this PR support for selecting internal subnets hasn't landed yet.

Also I'm told that my understanding of this is incomplete (by someone from the Lambda team), AND they're changing it early next year, AND we never explicate the reason on any other of our public documentation pages, we just say "no public for you". I don't want us to be the single source of truth for a factoid that's going to become wrong in a short while.